### ISSUE TITLE
Iterator corruption in leveldb::MemTableIterator::key 

### ISSUE THREAD

**Issue by slfritchie** [basho/eleveldb#102]
Same fault injection & test parameters & recreation steps as #101, with one small exception:  Change the final Erlang shell command to be:

```
[ok = eunit:test(iterators, [verbose]) || _ <- lists:seq(1,99999)].
```

A failing test case looks like this:

```
======================== EUnit ========================
module 'iterators'
  iterators: prev_test_case...[0.001 s] ok
  iterators: seek_and_next_test_case...ok
  iterators: basic_prefetch_test_case...ok
  iterators: seek_and_prefetch_test_case...*failed*
in function iterators:'-seek_and_prefetch_test_case/1-fun-6-'/2 (test/iterators.erl, line 93)
in call from iterators:'-seek_and_prefetch_test_case/1-fun-14-'/1 (test/iterators.erl, line 93)
**error:{assertEqual_failed,[{module,iterators},
                     {line,93},
                     {expression,"eleveldb : iterator_move ( I , << \"a\" >> )"},
                     {expected,{ok,<<"a">>,<<"w">>}},
                     {value,{ok,<<"d">>,<<"z">>}}]}
```

... which, as far as I can tell, shows the iterator skipping over key <<"a">>  as well as b & c and hopping directly to d.

**Comment by matthewvon** [comment]
retry with this:  https://github.com/basho/eleveldb/pull/104

**Comment by slfritchie** [comment]
Hrm, this problem appears that #104 has made this case either impossible with the fault injection method from #101 or makes it veryveryvery extremely unlikely.  I'll try again with some different faults.

Oh, the same applies for #101: I didn't get a single SEGV or bad iterator value, yay.

**Comment by matthewvon** [comment]
104 was a nasty bug.  The bug was the result of adding code to let AAE both skip around on a single iterator and use prefetch on same iterator.  I missed a timing dependency.  The bad code could let two threads call into leveldb with the same iterator object. 

Matthew Von-Maszewski

**Comment by slfritchie** [comment]
As far as I can tell, both the bad iterator of this ticket and crash of #101 are resolved by #104 and basho/leveldb#125
